test/cluster/test_hints.py

```
@@ -25,14 +27,82 @@
+
+
+def count_hint_segments(hints_dir: str, target_host_id: str, shard: int | None = None) -> int:
+    shard_glob = "*" if shard is None else str(shard)
+    return len(glob.glob(os.path.join(hints_dir, shard_glob, str(target_host_id), "HintsLog-*.log")))
+
 
-async def create_sync_point(client: TCPRESTClient, server_ip: IPAddress) -> str:
-    response = await client.post_json("/hinted_handoff/sync_point", host=server_ip, port=10_000)
+async def wait_until_hint_writing_settled(manager: ScyllaClusterManager, servers: list[ServerInfo],
+                                          timeout: float = 180) -> None:
+    async def check():
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `check`

(ANN202)